core/store/src/trie/trie_recording.rs

```diff
@@ -7,22 +7,31 @@ use std::sync::Arc;
 /// A simple struct to capture a state proof as it's being accumulated.
 pub struct TrieRecorder {
     recorded: HashMap<CryptoHash, Arc<[u8]>>,
+    size: usize,
 }
 
 impl TrieRecorder {
     pub fn new() -> Self {
-        Self { recorded: HashMap::new() }
+        Self { recorded: HashMap::new(), size: 0 }
     }
 
     pub fn record(&mut self, hash: &CryptoHash, node: Arc<[u8]>) {
-        self.recorded.insert(*hash, node);
+        let size = node.len();
+        if self.recorded.insert(*hash, node).is_none() {
+            self.size += size;
+        }
     }
```

Comment:
Is it possible to call `record()` again with the same `hash`, but different `node`?
If yes, then the new node might have a different size from the previous one and `self.size` should be adjusted.

Reply:
After talking with Anton he said that each node hash is unique, so it can't happen. There's no problem.